### ISSUE TITLE
MethodOverloading bug with nested overloaded methods

### ISSUE THREAD

**Issue by schalkms** [arturbosch/detekt#2247]
## Expected Behavior
The following code should only report the `Nested` class with a `threshold` set to 3.

```kotlin
class Outer {
  class Inner {
    fun f() {}
    fun f(i: Int) {}
    fun f(i: Int, i2: Int) {}
  }
}
```

## Observed Behavior
For the mentioned code sample the `MethodOverloading` rule reports 2 issues.
1. `Outer`
2. `Inner`

## Context
The rule must treat each class and it's containing functions as a single context by not looking deeper down the nested class tree.

## Your Environment

* Version of detekt used: 1.3.1
* Operating System and version: macOS 10.14.6
